numba_dppy/utils/array_utils.py

```diff
@@ -0,0 +1,205 @@
+# Copyright 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import dpctl
+import dpctl.memory as dpctl_mem
+
+supported_numpy_dtype = [
+    np.int32,
+    np.int64,
+    np.uint32,
+    np.int64,
+    np.float32,
+    np.float64,
+]
+
+
+def is_usm_backed(obj):
+    """
+    Determine and return a SYCL device accesible object.
+
+    Args:
+        obj: Object to be tested and data copied from.
+
+    Returns:
+        obj: USM backed object if object is USM backed, None otherwise.
+    """
+    usm_mem = None
+    try:
+        usm_mem = dpctl_mem.as_usm_memory(obj)
+    except:
+        if hasattr(obj, "base"):
+            try:
+                usm_mem = dpctl_mem.as_usm_memory(obj.base)
+            except Exception:
+                pass
+
+    return usm_mem
+
+
+def copy_to_usm_backed(usm_backed, obj):
+    """
+    Copy from supported objects to USM backed data.
```

Comment:
Add a detail doc section. It will be good to link to the `__sycl_usm_array_interface__` spec.